src/tags/object/Ranker.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * libraries
+ */
+import React from 'react';
+import { inject, observer } from 'mobx-react';
+import { flow, types } from 'mobx-state-tree';
+
+import Registry from '../../core/Registry';
+import { AnnotationMixin } from '../../mixins/AnnotationMixin';
+import ProcessAttrsMixin from '../../mixins/ProcessAttrs';
+import Base from './Base';
+import { parseValue } from '../../utils/data';
+
+/**
+ * components
+ */
+import Ranker from '../../components/Ranker/Ranker';
+import { getData } from '../../components/Ranker/createData';
+
+/**
+ * @name Ranker
+ * @meta_title Ranker Tag displays items that can be dragged and dropped between columns
+ * @meta_description Customize Label Studio by displaying and sorting results for machine learning and data science projects.
+ * @param {string} value Data field value containing JSON type for Ranker
+ * @param {string} [valueType] Value to define the data type in Ranker
+ */
+const Model = types
+  .model({
+    type: 'ranker',
+    value: types.maybeNull(types.string),
+    _value: types.frozen([]),
+    valuetype: types.optional(types.string, 'json'),
+    /**
+     * rank: 1 column, reorder to rank
+     * select: 2 columns, drag results to second column to select
+     */
+    mode: types.optional(types.enumeration(['rank', 'select']), 'select'),
+
+  })
+  .views(self => ({
+    get dataSource() {
+      return self._value;
+    },
+    get resultType() {
+      return 'ranker';
+    },
+    get valueType() {
+      return 'ranker';
+    },
+    get result() {
+      return self.annotation.results.find(r => r.from_name === self);
+    },
+  }))
+  .actions(self => ({
+    updateValue: flow(function* (store) {
+      const value = parseValue(self.value, store.task.dataObj);
+
+      //grabs data from the createData file
+      self._value = getData(value, self.mode);
+      yield Promise.resolve(true);
+    }),
+    needsUpdate() {
+      if (self.annotation && !self.result) {
+        self.annotation.createResult({}, { ranker: [] }, self, self);
+      }
```

Comment:
There can be added another case when we have `result` already, so we can preload already filtered/reordered items from it and alter `dataSource` by adding columns there (as I understand that's the main way to pass filtered state to your component)

Reply:
we do this in different ways in methods `updateFromResult`, `needsUpdate` and `setResult`